pwnlib/data/templates/pwnup.mako

```diff
@@ -131,13 +131,12 @@ def start(argv=[], *a, **kw):
 %endif
 gdbscript = '''
 %if ctx.binary:
+set sysroot
```

Comment:
IIRC, `set sysroot` with no arguments is a no-op.

`set sysroot /` is incorrect for SSH-forwarded GDB sessions and Android devices.

`set sysroot` will also break for foreign-architecture binaries which are emulated under QEMU:

https://github.com/Gallopsled/pwntools/blob/20cb04984ae9687de3f192c76e1cfa405b00665b/pwnlib/gdb.py#L424-L430

Finally, you don't want to set the sysroot in your GDB script.  Pass the `sysroot=` argument into the `gdb.debug()` or `gdb.attach()` call, that's what it's there for.  (Side bar, we don't actually invoke `set sysroot` for native-arch binaries, even if it's provided.  This is a small bug.)

Reply:
If I don't use `set sysroot` I get the following error:

```
Temporary breakpoint 1 at 0x605: file x.c, line 2.    
Warning:                                              
Cannot insert breakpoint 1.                           
Cannot access memory at address 0x605                 
                                                      
/tmp/pwnsIcbJL.gdb:7: Error in sourced command file:  
Command aborted.                                      
```

The right side is with `set sysroot`, the left without, both include `tbreak main` and `continue`:
![screenshot from 2018-05-10 19-25-24](https://user-images.githubusercontent.com/3750466/39883726-faa8c40e-5487-11e8-9768-1bc7f096e010.png)

Reply:
Can you run your script with `DEBUG` to see the actual `gdbscript` being executed? (i.e. `python foo.py LOCAL GDB DEBUG`).

Can you also provide the first line of `gdb --version` and all of `gdb --configuration`? Thanks!

Reply:
```
➜  tmp gdb --version
GNU gdb (GDB) 8.1
Copyright (C) 2018 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-pc-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word".
```

```
➜  tmp gdb --configuration
This GDB was configured as follows:
   configure --host=x86_64-pc-linux-gnu --target=x86_64-pc-linux-gnu
             --with-auto-load-dir=$debugdir:$datadir/auto-load
             --with-auto-load-safe-path=$debugdir:$datadir/auto-load
             --with-expat
             --with-gdb-datadir=/usr/share/gdb (relocatable)
             --with-jit-reader-dir=/usr/lib/gdb (relocatable)
             --without-libunwind-ia64
             --with-lzma
             --with-python=/usr (relocatable)
             --with-guile
             --with-separate-debug-dir=/usr/lib/debug (relocatable)
             --with-system-gdbinit=/etc/gdb/gdbinit
             --without-babeltrace

("Relocatable" means the directory can be moved with the GDB installation
tree, and GDB will still find it.)

```

```
➜  tmp gdbserver --version
GNU gdbserver (GDB) 8.1
Copyright (C) 2018 Free Software Foundation, Inc.
gdbserver is free software, covered by the GNU General Public License.
This gdbserver was configured as "x86_64-pc-linux-gnu"
```

Reply:
Here's the output of `./exploit.py GDB LOCAL DEBUG`

![screenshot from 2018-05-10 20-59-40](https://user-images.githubusercontent.com/3750466/39888431-134a133e-5495-11e8-82fb-7098ed47d74d.png)